schema/configuration-structure.asn

```diff
@@ -34,25 +34,114 @@ BEGIN
+         zhinxmin Zhinxmin OPTIONAL, -- Zhinxmin: Minimal Support for Half-Precision Floating-Point in Integer Registers X.  Implies Zfinx
+         zk     Zk     OPTIONAL,   -- Zk: Standard Scalar Cryptography.  Implies Zkn, Zkr, and Zkt
+         zkn    Zkn    OPTIONAL,   -- Zkn: NIST Cryptography, Implies Zbkb, Zbkc, Zbkx, Zkne, Zknd, and Zknh
+         zknd   Zknd   OPTIONAL,   -- Zknd: NIST Suite: AES Decryption.
+         zkne   Zkne   OPTIONAL,   -- Zkne: NIST Suite: AES Encryption.
+         zknh   Zknh   OPTIONAL,   -- Zknh: NIST Suite: SHA2 Hashing.
+         zkr    Zkr    OPTIONAL,   -- Zkr: Entropy Source Extension
+         zks    Zks    OPTIONAL,   -- Zks: ShangMi Algorithm Cryptography.  Implies Zbkb, Zbkc, Zbkx, Zksed, and Zksh
+         zksed  Zksed  OPTIONAL,   -- Zksed: ShangMi Suite: SM4 Block Cipher.
+         zksh   Zksh   OPTIONAL,   -- Zksh: ShangMi Suite: SM3 Hashing.
+         zkt    Zkt    OPTIONAL,   -- Zkt: Data Independent Execution Latency
+         ...
+      } OPTIONAL, -- stdUnprivExtensions
+
+      stdSupLevelExtensions SEQUENCE {
+         sm1p11 Sm1p11 OPTIONAL,   -- Sm1p11: Machine Architecture v1.11
+         sm1p12 Sm1p12 OPTIONAL,   -- Sm1p12: Machine Architecture v1.12
+         smepmp Smepmp OPTIONAL,   -- Smepmp: PMP Enhancements for memory access and execution prevention on Machine mode
+         smstateen Smstateen OPTIONAL, -- Smstateen: State-Enable
+         ss1p11 Ss1p11 OPTIONAL,   -- Ss1p11: Supervisor Architecture v1.11
+         ss1p12 Ss1p12 OPTIONAL,   -- Ss1p12: Supervisor Architecture v1.12
+         sscofpmf Sscofpmf OPTIONAL, -- Sscofpmf: Count Overflow and Mode-Based Filtering
+         sstc   Sstc   OPTIONAL,   -- Sstc: Supervisor-mode Timer Interrupts
+         sv32   Sv32   OPTIONAL,   -- Sv32: Page-based Virtual Memory, 32-bit
+         sv39   Sv39   OPTIONAL,   -- Sv39: Page-based Virtual Memory, 39-bit
+         sv48   Sv48   OPTIONAL,   -- Sv48: Page-based Virtual Memory, 48-bit
+         sv57   Sv57   OPTIONAL,   -- Sv57: Page-based Virtual Memory, 57-bit
+         svinval Svinval OPTIONAL, -- Svinval: Fine-Grained Address-Translation Cache Invalidation
+         svnapot Svnapot OPTIONAL, -- Svnapot: NAPOT Translation Contiguity
+         svpbmt Svpbmt OPTIONAL,   -- Svpbmt: Page-Based Memory Types
+      } OPTIONAL, -- stdSupLevelExtensions
+
+      stdHypLevelExtensions SEQUENCE {
+         h H OPTIONAL, -- H: Hypervisor
+      } OPTIONAL, -- stdHypLevelExtensions
+
+      stdMachLevelExtensions SEQUENCE {
+      } OPTIONAL, -- stdMachLevelExtensions
+
+      nonStdExtensionsExist Boolean,
+      nonStdExtensions SEQUENCE OF {
+         nonStdExtension PrintableString, -- Prefixed with a single “X”, ordered alphabetically and separated from other multi-letter extension names by an underscore on the RISC-V ISA name strings 
+         nonStdExtensionVersion Version OPTIONAL
+      } OPTIONAL, -- nonStdExtensions
```

Comment:
This doesn't include a feature for adding extra information for the non-standard extension. The Custom type tries to do the same thing you did here, for vendor extensions. It includes some black box data for the custom feature to use, if they want.

Almost certainly it's better to define a type (including its name) for non-standard extensions, and then here to make a SEQUENCE OF that type. Putting all the non-standard extensions together into a single string seems unnecessarily complex to deal with for software when structured data is readily available.

Reply:
Thanks for the feedback @timsifive .  I have deleted nonStdExtensions, and added name and version fields to the Custom SEQUENCE.